### ISSUE TITLE
Log Retry and Connection in same Qlog file, use ODCID for indentification

### ISSUE THREAD

**Issue by huitema** [quicwg/qlog#105]
Many implementations identify the Qlog file using the "Initial DCID" used by the client. In the case of Retry, this results in two Qlog traces: one identified by the original DCID that contains the first initial packet of the client and the Retry packet that the server produced, and a second one identified by the Initial DCID used in the subsequent connection.

My proposal is to use a single file for the retry and for the next connection, identified by the ODCID. But I wonder whether that's going to break something.

**Comment by huitema** [comment]
The current spec kind of gives up on the issue. It says "Note that this can make it difficult to match logs from different vantage points with each other. For example, from the client side, it is easy to log connections with version negotiation or stateless retry in the same trace, while on the server they would most likely be logged in separate traces." My point is that it can be solved on the server, if the server keys logging events with the ODCID.

**Comment by marten-seemann** [comment]
> My point is that it can be solved on the server, if the server keys logging events with the ODCID.

That's correct, and it's a trivial thing to do. The server needs to encode the ODCID in the token anyway, to be able to send it in the transport parameters. In fact, that's how quic-go does it: https://github.com/lucas-clemente/quic-go/blob/e7fa420e264c4d319625c6dbf7cdf1c7668e7535/server.go#L452-L456

**Comment by rmarx** [comment]
@huitema I'm not 100% sure what the issue is here.

The current qlog text explicitly mentions using the ODCID as filename for qlog files as the "lowest common denominator". Furthermore, qlog files can contain multiple individual traces, as the `traces' property is an array. So it's already perfectly possible to log one trace for the first connection and then a second trace for the retried one after that. As @marten-seemann said, there are people doing just that already. Maybe the problem is in the wording? That you interpret a `trace' as a single file, and I interpret it as a list of events, where 1 file can contain multiple traces? 

Another approach would be to use the `group_id' field and not set that to an ODCID but assign that a sort of internal server number based on IP+port or something to uniquely identify a client across attempts. That would work even for "version negotiation", as there you don't have the luxury of having a stable ODCID. I don't want to mandate this type of thing in the text though, since it all implies the server needs to keep extra state around to match up connections, and some (most?) won't want to do that (in production). 

In short: I'm not sure what you'd like me to change to the text at this time.

**Comment by marten-seemann** [comment]
> So it's already perfectly possible to log one trace for the first connection and then a second trace for the retried one after that. As @marten-seemann said, there are people doing just that already.

Actually, that's not what I'm doing. I'm not logging the first Intitial received and the Retry sent at all on the server side (only on the client side), since there's no reasonable way of logging events that happen outside of a connection (see #106).

**Comment by LPardue** [comment]
I might suggest that the people that care about logging Retry are those using it for DoS mitigations, and they have a whole bunch of other factors to consider too.

Let's see if the current text can be minimally changed for clarity. If someone with the problem wants to propose a concrete suggestion that differs from this in a PR that would be great but lets not sit around forever in anticipation.

**Comment by marten-seemann** [comment]
I would claim that this issue was resolved by https://github.com/quicwg/qlog/pull/279. Unless anybody objects within the next 24h, I'm going to close this issue and the associated PR (#213).
